web/assets/js/components/gameboard.ts

```diff
@@ -165,6 +165,18 @@ const Gameboard = class extends Component {
 
       //TODO: chech if this.getMoveNumber() return 0 or 1 for before first move.
       if (this.getMoveNumber() > 1) {
+        const dateStartGame = gameState.time?.started_at;
+
+        if (!dateStartGame) {
+          throw new Error('No started_at game time');
+          return;
+        }
+        // Get the time of creation & current time
+        // Get the diff and use it to get the updated / sync timeout by substract by 30s
+        const startDate = new Date(dateStartGame);
```

Comment:
Do we need to check that startData < current date (time)?

Reply:
@zivkovicmilos I don't think this is possible cause it would mean the front has started the game before the server but we need the game to be ready to leave the lobby to enter the party right? I mean the `started_at` will be created with the game right @thehowl ?

However we might have a timer that is > 30s (imagine a 30s lag lol) OK that's hard to get but who knows...

```js

    const exitforTimeout = async () => {
       try {
          // Claim timeout. If no error, timeout succeeded
          await actions.claimTimeout(this.gameId);
          this.call(this.engine(false, 'timeout'));
        } catch (e) {
           this.call(this.engine(false, 'timeout'));  //TODO: create a fail "exit" gameover in the engine
        }
     };

     const startDate = new Date(dateStartGame);
     const currentDate = new Date();
     const diffDate = startDate.getTime() - currentDate.getTime();
     
     const timer = 30 * 1000 - diffDate;

     if (timer <= 0) {
        // if date too old (> 30sec )
        await exitforTimeout();
        throw new Error('Game party started too long time ago');
     }

Reply:
>Do we need to check that startData < current date (time)?

I don't think so, it should only be able to happen if the user's system date was wrong 

>I mean the started_at will be created with the game right @thehowl ?

Yes